internal/provider/providers/scaleway/provider.go

```diff
@@ -0,0 +1,188 @@
+package scaleway
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/provider/constants"
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/headers"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	domain 		string
+	owner  		string
+	ipVersion  	ipversion.IPVersion
+	ipv6Suffix 	netip.Prefix
+	secretKey   string
+	field_name	string
+	ttl		 	uint16
+}
+
+func New(data json.RawMessage, domain, owner string,
+	ipVersion ipversion.IPVersion, ipv6Suffix netip.Prefix) (
+	provider *Provider, err error,
+) {
+	var providerSpecificSettings struct {
+        SecretKey	string	`json:"secret_key"`
+		FieldName	string	`json:"field_name"`
+		TTL			uint16	`json:"ttl"`
+    }
+	err = json.Unmarshal(data, &providerSpecificSettings)
+	if err != nil {
+		return nil, fmt.Errorf("json decoding provider specific settings: %w", err)
+	}
+
+	if providerSpecificSettings.TTL == 0 {
+        providerSpecificSettings.TTL = 3600
+    }
+
+	err = validateSettings(domain,
+		providerSpecificSettings.SecretKey)
+	if err != nil {
+		return nil, fmt.Errorf("validating provider specific settings: %w", err)
+	}
+
+	return &Provider{
+		domain:     domain,
+		owner:      owner,
+		ipVersion:  ipVersion,
+		ipv6Suffix: ipv6Suffix,
+        secretKey:  providerSpecificSettings.SecretKey,
+		field_name: providerSpecificSettings.FieldName,
+		ttl:		providerSpecificSettings.TTL,
+	}, nil
+}
+
+func validateSettings(domain, secretKey string) (err error) {
+	err = utils.CheckDomain(domain)
+	if err != nil {
+		return fmt.Errorf("%w: %w", errors.ErrDomainNotValid, err)
+	}
+
+	switch {
+    case secretKey == "":
+        return fmt.Errorf("%w", errors.ErrSecretKeyNotSet)
+    }
```

Comment:
since there is only one case

```suggestion
	if secretKey == "" {
        return fmt.Errorf("%w", errors.ErrSecretKeyNotSet)
    }
```